binary_search_tree/tree.py

```diff
@@ -14,44 +18,179 @@ class Tree:
+            return current_root.value
+    
     def find(self, key):
-        pass
+        if self.root == None:
+            return None
+        
+        result = self.find_helper(self.root, key)
+        return result
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
+    def inorder_helper(self, current_node, items):
+        if current_node != None:
+            self.inorder_helper(current_node.left, items)
+            items.append({"key": current_node.key, "value": current_node.value})
+            self.inorder_helper(current_node.right, items)
+    
     def inorder(self):
-        pass
+        items = []
 
-    # Time Complexity: 
-    # Space Complexity:     
+        if self.root == None:
+            return items
+
+        self.inorder_helper(self.root, items)
+        
+        return items
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)   
+    def preorder_helper(self, current_node, items):
+        if current_node != None:
+            items.append({"key": current_node.key, "value": current_node.value})
+            self.preorder_helper(current_node.left, items)
+            self.preorder_helper(current_node.right, items)
+            
+    
     def preorder(self):
-        pass
+        items = []
+
+        if self.root == None:
+            return items
+        
+        self.preorder_helper(self.root, items)
+
+        return items
 
     # Time Complexity: 
     # Space Complexity:     
+    def postorder_helper(self, current_node, items):
+        if current_node != None:
+            self.postorder_helper(current_node.left, items)
+            self.postorder_helper(current_node.right, items)
+            items.append({"key": current_node.key, "value": current_node.value})
+            
+    
     def postorder(self):
-        pass
+        items = []
 
-    # Time Complexity: 
-    # Space Complexity:     
+        if self.root == None:
+            return items
+        
+        self.postorder_helper(self.root, items)
+
+        return items
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)  
+    def height_helper(self, node):
+        if node == None:
+            return 0
+        
+        left_height = self.height_helper(node.left)
+        right_height = self.height_helper(node.right)
+
+        return 1 + max(left_height, right_height)
+    
     def height(self):
-        pass
+        if self.root == None:
+            return 0
+
+        current_node = self.root
+
+        tree_height = self.height_helper(current_node)
+
+        return tree_height
 
 
 #   # Optional Method
-#   # Time Complexity: 
-#   # Space Complexity: 
+#   # Time Complexity: O(n)
+#   # Space Complexity: O(n)
     def bfs(self):
-        pass
+        items = []
+        
+        if self.root == None:
+            return items
+        
+        items = [self.root]
+        queue = [{"key": self.root.key, "value": self.root.value}]
```

Comment:
This works, but you can also just use the nodes themselves instead of making a new dictionary.